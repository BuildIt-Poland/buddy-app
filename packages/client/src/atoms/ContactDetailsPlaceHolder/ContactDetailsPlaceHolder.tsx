import React from 'react';
import makeStyles from '@material-ui/core/styles/makeStyles';
import Box from '@material-ui/core/Box';
import Skeleton from '@material-ui/lab/Skeleton';

const useStyles = makeStyles(theme => ({
  subHeader: {
    height: '2.5em',
    width: '40%',
  },
  avatar: {
    height: '6rem',
    width: '6rem',
    margin: theme.spacing(1, 2, 0, 1),
  },
  info: {
    height: '2.5rem',
    width: '15rem',
  },
  wrapper: {
    marginTop: theme.spacing(2),
  },
}));

const ContactDetailsPlaceHolder: React.FC = () => {
  const classes = useStyles();

  const PlaceHolderContactItem = () => (
    <>
      <Skeleton variant='text' className={classes.subHeader} />
      <Skeleton variant='text' className={classes.info} />
    </>
  );

  return (
    <Box className={classes.wrapper} display='flex'>
      <Skeleton variant='circle' className={classes.avatar} />
      <Box>
        <PlaceHolderContactItem />
        <PlaceHolderContactItem />
        <PlaceHolderContactItem />
        <PlaceHolderContactItem />
      </Box>
    </Box>
  );
};

export default ContactDetailsPlaceHolder;
