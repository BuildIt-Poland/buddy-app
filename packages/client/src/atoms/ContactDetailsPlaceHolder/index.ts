import ContactDetailsPlaceHolder from './ContactDetailsPlaceHolder';

export default ContactDetailsPlaceHolder;
