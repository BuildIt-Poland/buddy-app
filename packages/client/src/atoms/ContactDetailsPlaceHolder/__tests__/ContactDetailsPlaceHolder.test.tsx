import React from 'react';
import { create } from 'react-test-renderer';

import ContactDetailsPlaceHolder from '../ContactDetailsPlaceHolder';

jest.mock('@material-ui/lab/Skeleton', () => 'Skeleton');
jest.mock('@material-ui/core/Box', () => 'Box');

describe('Component - ContactDetailsPlaceHolder', () => {
  test('renders correctly', () => {
    const component = create(<ContactDetailsPlaceHolder />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
