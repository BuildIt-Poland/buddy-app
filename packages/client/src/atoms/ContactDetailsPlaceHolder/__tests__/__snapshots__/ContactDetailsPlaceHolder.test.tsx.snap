// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - ContactDetailsPlaceHolder renders correctly 1`] = `
<Box
  className="makeStyles-wrapper-4"
  display="flex"
>
  <Skeleton
    className="makeStyles-avatar-2"
    variant="circle"
  />
  <Box>
    <Skeleton
      className="makeStyles-subHeader-1"
      variant="text"
    />
    <Skeleton
      className="makeStyles-info-3"
      variant="text"
    />
    <Skeleton
      className="makeStyles-subHeader-1"
      variant="text"
    />
    <Skeleton
      className="makeStyles-info-3"
      variant="text"
    />
    <Skeleton
      className="makeStyles-subHeader-1"
      variant="text"
    />
    <Skeleton
      className="makeStyles-info-3"
      variant="text"
    />
    <Skeleton
      className="makeStyles-subHeader-1"
      variant="text"
    />
    <Skeleton
      className="makeStyles-info-3"
      variant="text"
    />
  </Box>
</Box>
`;
